libglusterfs/src/dict.c

```diff
@@ -376,24 +375,12 @@ dict_set_lk(dict_t *this, char *key, const int key_len, data_t *value,
         }
     }
 
-    if (this->free_pair.key) { /* the free_pair is used */
-        pair = mem_get(THIS->ctx->dict_pair_pool);
-        if (!pair) {
-            return -1;
-        }
-    } else { /* assign the pair to the free pair */
-        pair = &this->free_pair;
-    }
-
-    pair->key = (char *)GF_MALLOC(key_len + 1, gf_common_mt_char);
-    if (!pair->key) {
-        if (pair != &this->free_pair) {
-            mem_put(pair);
-        }
+    pair = GF_MALLOC(sizeof(data_pair_t) + key_len + 1, gf_common_mt_char);
```

Comment:
I think a `gf_common_mt_data_pair_t` type would be more useful here.

Reply:
there's no such type - do you mean adding a new one?

Reply:
Yes. `gf_common_mt_char` is a very generic type. Since this is a very specific structure used for a very specific purpose, labeling it with a specific type will be better. This is useful to help identify things during debugging.